src/pages/Results.tsx

```diff
@@ -0,0 +1,119 @@
+import { FC, useEffect, useState, useContext } from "react";
+import { FhirContext } from "../components/fhir/FhirContext";
+
+import LoadingSpinner from "../components/UI/LoadingSpinner";
+import ModalWrapper from "../components/UI/ModalWrapper";
+import { ModalState } from "../components/UI/ModalWrapper";
+import ResultsList from "../components/results-list/ResultsList";
+
+const FHIR_URL = process.env.REACT_APP_FHIR_URL;
+
+type patientResult = {
+  firstName: string;
+  lastName: string;
+  patientId: string;
+  observations: { [key: string]: any }[]; // using any since it is a deeply nested object from an external api}
+};
+export type parsedResultsModel = patientResult[];
+
+const Results: FC = () => {
+  const ctx = useContext(FhirContext);
+  const [modal, setModal] = useState<ModalState | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [parsedResults, setParsedResults] = useState<parsedResultsModel | null>(null);
+
+  useEffect(() => {
+    const observationQueryUrl = `${FHIR_URL}/Observation?_profile=http://hl7.org/fhir/uv/genomics-reporting/StructureDefinition/variant&_include=Observation:subject`;
+
+    setIsLoading(true);
+
+    const requestObservations = async () => {
+      ctx.client
+        ?.request(observationQueryUrl)
+        .then((response) => {
+          const parsedResults = parseResults(response.entry);
+          setParsedResults(parsedResults);
+        })
+        .catch((error) => {
+          setModal({
+            message: "Something went wrong fetching observations from the FHIR server.",
+            isError: true,
+          });
+          console.error(error);
+        })
+        .finally(() => {
+          setIsLoading(false);
+        });
+    };
+    requestObservations();
+  }, [ctx]);
+
+  const parseResults = (entries: { [key: string]: any }[]) => {
+    console.log(entries);
+    // extract patients from the data
+    const patients = entries.filter((entry: { [key: string]: any }) => {
+      return entry.fullUrl.includes("Patient");
+    });
+
+    // extract observations from the data
+    const observations = entries.filter((entry: { [key: string]: any }) => {
+      return entry.fullUrl.includes("Observation");
+    });
+
+    const readableResults = createReadableResults(patients, observations);
+    return readableResults;
+  };
+
+  const createReadableResults = (patients: { [key: string]: any }[], observations: { [key: string]: any }[]) => {
```

Comment:
The @smile-cdr library has the response types for a bundle so we could use these, along with `Patient` and `Observation`. May need to define one response type from a search set but feels like it should be fairly minimal typing to get there, and allows us to be intentional about what fields we have to check for being defined. 

So I think something like this (though you'll need to edit the rest of the function to directly use the array of resources

```suggestion  const parseResults = (entries: BundleEntry[]) => {
    console.log(entries);
    // extract patients from the data
    const patients = entries
      .filter((entry) => {
        return entry.fullUrl?.includes("Patient");
      })
      .map((entry) => entry.resource as Patient);

    // extract observations from the data
    const observations = entries
      .filter((entry) => {
        return entry.fullUrl?.includes("Observation");
      })
      .map((entry) => entry.resource as Observation);

    const readableResults = createReadableResults(patients, observations);
    return readableResults;
  };

  const createReadableResults = (patients: Patient[], observations: Observation[]) => {
```